Mainnet/Identity/IdentityContracts/IdentityProvider.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Stratis.SmartContracts;
+
+/// <summary>
+/// A constrained version of ERC780.
+/// This contract only holds claims from one identity provider, the "Owner".
+/// </summary>
+public class IdentityProvider : SmartContract
+{
+    private Address Owner
+    {
+        get
+        {
+            return this.PersistentState.GetAddress(nameof(Owner));
+        }
+        set
+        {
+            this.PersistentState.SetAddress(nameof(Owner), value);
+        }
+    }
+
+    public IdentityProvider(ISmartContractState state) : base(state)
+    {
+        this.Owner = Message.Sender;
+    }
+
+    public void ChangeOwner(Address newOwner)
+    {
+        Assert(this.Owner == Message.Sender);
+        this.Owner = newOwner;
+    }
+
+    public void AddClaim(Address issuedTo, uint topic, byte[] data)
+    {
+        Assert(this.Owner == Message.Sender);
+
+        byte[] oldData = this.GetClaim(issuedTo, topic);
+        bool exists = oldData != null;
+
+        // This exact claim value is stored already. Nothing to change.
+        if (exists && ByteArrayCompare(oldData, data))
+            return;
+
+        this.SetClaim(issuedTo, topic, data);
+
+        if (exists)
+        {
+            this.Log(new ClaimChanged
+            {
+                IssuedTo = issuedTo,
+                Topic = topic,
+                Data = data
+            });
+        }
+        else
+        {
+            this.Log(new ClaimAdded
+            {
+                IssuedTo = issuedTo,
+                Topic = topic,
+                Data = data
+            });
+        }
```

Comment:
Nit pick, also depends on current internal use cases that may already exist but...

These logs for `ClaimChanged, ClaimAdded, and ClaimRemoved` all share the same shape but since they are different, querying this contracts receipts would require a separate Cirrus Local Call for each event to check the logs of all 3 types. Not breaking but more difficult to use in my experience.

**Suggestion**
Use 1 new log type of `ClaimEvent`, adding a new property for `Type` and Index that in the event.

``` C#
// New single event type
public struct ClaimEvent
{
    [Index]
    public Address IssuedTo;
    [Index]
    public uint Topic;
    public byte[] Data;
    [Index]
    public ClaimType Type;
}

// Type enum 
public enum ClaimType : int
{
    Added = 0,
    Changed = 1,
    Removed = 2
}

// Log accordingly
this.Log(new ClaimAdded
{
    IssuedTo = issuedTo,
    Topic = topic,
    Data = data,
    Type = ClaimType.Added // or any other claim type
});
```

Reply:
On receipt/search endpoint, we specify log filter by name so it is important to differentiate log names.

Reply:
You can have 1 `ClaimEvent` that returns all claim types and just filter the JSON returned for the ones you wanted. otherwise if you ever wanted all types, you'll have to query for each type individually. Up to you guys, sometimes easier to combine like shaped objects.